### PR TITLE
GIPF-81/ place a piece

### PR DESCRIPTION
**Pull request by sulphite**
- [x] provide websocket context
- [x] on click, send coordinate to backend
- [x] send response with validity of move
- [x] handle response, highlight possible moves
- [x] can't click inner hexes

Scope of this is getting too large. Getting the piece to be the correct colour is related to turn structure and will be fixed in a separate ticket.

### CODE REVIEW

File: frontend/src/components/Hexagon.css [hunk 1]
```diff
@@ -22,3 +32,18 @@
...
+@keyframes pulse {
+  0% {
+    fill:rgb(113, 242, 214);
+  }
+
+  50% {
+    fill:rgb(146, 255, 231);
+  }
+
+
```
Comment (adrianHards) [inline frontend/src/components/Hexagon.css:45]: a lot of empty lines; would also encourage everything to be ordered alphabetically

File: frontend/src/components/Hexagon.tsx [hunk 2]
```diff
@@ -1,23 +1,48 @@
...
+      }
+    } else if(clickable) {
+      try {
+        console.log(`sending ${data.coords} to backend`)
+        handleSendMove(data.coords)
+      } catch (error: unknown) {
+        console.error(error)
+      }
+    } else {
+      console.log("clicked inner hex")
```
Comment (adrianHards) [inline frontend/src/components/Hexagon.tsx:39]: what's the eventual purpose of line 39?

File: backend/src/logic/Game.ts [hunk 3]
```diff
@@ -26,21 +26,25 @@ export class Game {
...
       .filter((tile) => this.board.isPushable(this.board.getRow(coord, tile)))
       .toArray();
-    if (legalTiles.length > 0) {
-      this.board.fillTile(coord, player);
-      this.score[player] -= 1;
-    }
+    // if (legalTiles.length > 0) {
+    //   this.board.fillTile(coord, player);
+    //   this.score[player] -= 1;
+    // }
```
Comment (adrianHards) [inline backend/src/logic/Game.ts:39]: does this need to be kept?

**Review by adrianHards — APPROVED** [review]
see comments

File: frontend/src/components/Hexagon.css (hunk 1 above)
Comment (sulphite) [reply frontend/src/components/Hexagon.css:45]: not sure alphabetical is the most intuitive way? But there should be some kind of ordering

File: frontend/src/components/Hexagon.tsx (hunk 2 above)
Comment (sulphite) [reply frontend/src/components/Hexagon.tsx:39]: eventually there will need to be some logic to handle move resolutions using these inner tiles

**Pull request merged**